File: liqp-core/src/main/java/liqp/nodes/NEqNode.java
package liqp.nodes;

import liqp.node.LNode;
import liqp.node.LValue;
import lombok.Getter;

@Getter
public class NEqNode extends ExpressionNode {

  public NEqNode(LNode lhs, LNode rhs) {
    super(lhs, rhs);
  }

  @Override
  public Object render(LContext context) {

    Object a = lhs.render(context);
    Object b = rhs.render(context);

    return !LValue.Companion.areEqual(a, b);
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/NEqNode.kt
package liqp.nodes

import liqp.ComparisonResult.GREATER
import liqp.ComparisonResult.LESS
import liqp.ComparisonResult.NOOP
import liqp.ComparisonResult.NULL
import liqp.node.LNode

class NEqNode(lhs: LNode, rhs: LNode) : ComparisonNode(lhs, rhs) {
  override val matchingComparisons = setOf(NULL, LESS, GREATER, NOOP)
}
